### PR TITLE
strictyaml: Allow migration errors.

### PR DESCRIPTION
**Pull request by daavoo**
Background in #9630

### CODE REVIEW

**Review by dberenbaum — APPROVED** [review]
No objection from me as a hotfix. Thanks!

**Comment by dberenbaum** [comment]
Can be a follow-up PR, but can we leave in a warning about those errors instead of ignoring them completely?

Also, can we handle them consistently? It looks like they will sometimes be raised as errors and sometimes be ignored, depending on whether there are other errors.

File: dvc/utils/strictyaml.py [hunk 1]
```diff
@@ -254,18 +254,32 @@ def __pretty_exc__(self, **kwargs: Any) -> None:
...
+        error.msg == "extra keys not allowed"
+        and error.path[0] == "stages"
+        and (
+            error.path[2] == "vars"
+            or (error.path[2] == "outs" and error.path[5] == "checkpoint")
+        )
+    )
+
+
 def validate(
```
Comment (skshetry) [inline dvc/utils/strictyaml.py:268]: This is a helper method. This is not the place to handle these.

**Comment by skshetry** [comment]
Let's `monkeypatch` in Studio unless it's decided in dvc.

**Pull request closed without merging**